### ISSUE TITLE
jstree and IE11

### ISSUE THREAD

**Issue by MariusRoe** [vakata/jstree#418]
I am having a problem with IE 11, jstree works fine in all the other browsers - loading from ajax. but in IE 11 i just get the loading spinner forever. I have tried the pre1.0 release as well as the 2.0.0alpha release with the same results

**Comment by clasie** [comment]
Hello,

   Nice tree, thank you for that.

   I go the same issue only in ie11 for me.

   Thanks for any help on this.

Version jstree: /*! jsTree - v3.3.1 - 2016-05-05 - (MIT) */

Tree config
----------------------------------
       function buildJsTree(testData) {
            $('#jstree_demo')
                .jstree({
                    "core": {
                        "check_callback": function (op, node, node_parent) {
                            return op == 'move_node' ? node_parent.data.TYPE == "QUESTIONNAIRE" : true;
                        },
                        "animation": 500,
                        //"check_callback": true,
                        "themes": { "stripes": true },
                        "dataType": "xml",
                        'data': testData
                    },
                    "dnd": {
                        is_draggable: function (x) {
                            console.log("from is_draggable: " + x[0].data.TYPE);
                            return x[0].data.TYPE == "QUESTION";
                        },
                        drop_finish: function (data) {
                            console.log("eee");
                        },
                    },
                    "types": {
                        "#": {"max_children": -1, "max_depth": 4, "valid_children": ["root"]},
                        "root": {"icon": "/static/3.0.2/assets/images/tree_icon.png", "valid_children": ["default"]},
                        "default": {"valid_children": ["default", "file"]},
                        "file": {"icon": "glyphicon glyphicon-file", "valid_children": []}
                    },
                    "plugins": [/*"contextmenu",  "dnd", */ "dnd", "search", "state", "types", "wholerow"]
                }).bind('move_node.jstree', function (e, data) {
                    var dataToTransfer = {
                        service: 'REORDER_DRAGGED_QUESTION',
                        position: data.position,
                        old_position: data.old_position,
                        parent: data.parent,
                        old_parent: data.old_parent,
                        id_question: data.node.id
                    }
                });
        }
Data put in jstree
----------------------------------
{"jsTreeNodes":{"nodes":[{"id":"181","parent":"#","text":"Recette mayonnaise","data":{"TYPE":"QUESTIONNAIRE","ETAT":"2","SAUVEGARDE":"1","EXAMEN":"1","DATE_START":"04\/12\/2015","DATE_STOP":"24\/12\/2015","AUTO_CORRECTION":"1","COMMENT":"Recette mayonnaise","INTITULE":"","TYPE_REPONSE":"0","TYPE_VALUES":null,"IMAGES":[{"file_url":"..\/..\/..\/upload\/53\/questionnaire_181_potdemayo.jpg","file_name":"questionnaire_181_potdemayo.jpg","file_error":"","file_success":"","file_type":"","file_size":"","file_temp":"","file_id":"68"}],"ticket":{"id":"88","value":"q1q-MOt-qkN-Cn4-Gd9-1tma","label":""},"isChrono":"off","seconds":"60","points":null,"campaigns":[{"title":"Recette mayonnaise \/id\/188","id":"188","status":"1"}],"GOOD_RESPONSE_TXT":"","GOOD_RESPONSE_DATE":"00\/00\/0000","GOOD_RESPONSE_NUMBER":"","INCLUSIVE_TYPE_VALUES":[],"EXCLUSIVE_TYPE_VALUES":[]}},{"id":"182","parent":"181","text":"Question nombre d'oeufs","data":{"TYPE":"QUESTION","ETAT":null,"SAUVEGARDE":null,"EXAMEN":null,"DATE_START":null,"DATE_STOP":null,"AUTO_CORRECTION":null,"COMMENT":null,"INTITULE":"<div>Une <b>\u00e9mulsion <\/b>d'huile, voil\u00e0 ce qu'est la <span style=\"color:rgb(255,153,0);\"><b style=\"color:rgb(255,153,0);\">mayonnaise<\/b><\/span>. Une simple r\u00e9action chimique qui consiste \u00e0 emprisonner des gouttes d'huile dans un liant : le jaune d\u2019\u0153uf. Les autres ingr\u00e9dients sont l\u00e0 pour aromatiser : moutarde, vinaigre, jus de citron..<\/div><div><br><\/div><div>Question: pour 4 personnes combien de jaune d\u2019\u0153ufs?<\/div>","TYPE_REPONSE":"2","TYPE_VALUES":null,"IMAGES":[{"file_url":"..\/..\/..\/upload\/53\/question_182_oeufs.jpg","file_name":"question_182_oeufs.jpg","file_error":"","file_success":"","file_type":"","file_size":"","file_temp":"","file_id":"69"}],"ticket":{"id":null,"value":null,"label":null},"isChrono":null,"seconds":null,"points":"10","campaigns":"","GOOD_RESPONSE_TXT":"","GOOD_RESPONSE_DATE":"00\/00\/0000","GOOD_RESPONSE_NUMBER":"1","INCLUSIVE_TYPE_VALUES":[],"EXCLUSIVE_TYPE_VALUES":[]}},{"id":"183","parent":"181","text":"L'huile","data":{"TYPE":"QUESTION","ETAT":null,"SAUVEGARDE":null,"EXAMEN":null,"DATE_START":null,"DATE_STOP":null,"AUTO_CORRECTION":null,"COMMENT":null,"INTITULE":"<div>Les huiles v\u00e9g\u00e9tales ont cette propri\u00e9t\u00e9 naturelle de faire ressortir le go\u00fbt des autres aliments, tout simplement parce que beaucoup d\u2019ar\u00f4mes et de go\u00fbts sont port\u00e9s par des mol\u00e9cules dites lipophiles (qui se lient aux lipides).<\/div><div><br><\/div><div><b>Question<\/b>: quantit\u00e9 d'huile pour une mayonnaise &nbsp;pour 4 personnes?<\/div><div><br><\/div>","TYPE_REPONSE":"4","TYPE_VALUES":null,"IMAGES":[{"file_url":"..\/..\/..\/upload\/53\/question_183_huile.jpg","file_name":"question_183_huile.jpg","file_error":"","file_success":"","file_type":"","file_size":"","file_temp":"","file_id":"71"}],"ticket":{"id":null,"value":null,"label":null},"isChrono":null,"seconds":null,"points":"","campaigns":"","GOOD_RESPONSE_TXT":"","GOOD_RESPONSE_DATE":"00\/00\/0000","GOOD_RESPONSE_NUMBER":"","INCLUSIVE_TYPE_VALUES":[],"EXCLUSIVE_TYPE_VALUES":[{"text":"5 cl","isGood":false,"type":null,"points":"0","id_qcm_response":"238"},{"text":"10 cl","isGood":true,"type":null,"points":"10","id_qcm_response":"239"},{"text":"15 cl","isGood":false,"type":null,"points":"-5","id_qcm_response":"240"}]}},{"id":"184","parent":"181","text":"Moutarde","data":{"TYPE":"QUESTION","ETAT":null,"SAUVEGARDE":null,"EXAMEN":null,"DATE_START":null,"DATE_STOP":null,"AUTO_CORRECTION":null,"COMMENT":null,"INTITULE":"<div><b>M\u00eame<\/b> si elle est davantage connue comme un condiment, la moutarde est une plante dont les graines et les feuilles sont comestibles\u2026 et sources de nombreux bienfaits ! DocteurBonneBouffe vous livre tous les secrets de la moutarde.<\/div><div><br><\/div><div><b>Question<\/b>: quantit\u00e9 de moutarde forte pour une mayonnaise &nbsp;pour 4 personnes?<\/div>","TYPE_REPONSE":"4","TYPE_VALUES":null,"IMAGES":[{"file_url":"..\/..\/..\/upload\/53\/question_184_moutarde.jpg","file_name":"question_184_moutarde.jpg","file_error":"","file_success":"","file_type":"","file_size":"","file_temp":"","file_id":"72"}],"ticket":{"id":null,"value":null,"label":null},"isChrono":null,"seconds":null,"points":"","campaigns":"","GOOD_RESPONSE_TXT":"","GOOD_RESPONSE_DATE":"00\/00\/0000","GOOD_RESPONSE_NUMBER":"","INCLUSIVE_TYPE_VALUES":[],"EXCLUSIVE_TYPE_VALUES":[{"text":"1 cuill\u00e8re \u00e0 caf\u00e9 de moutarde forte","isGood":false,"type":null,"points":"-2","id_qcm_response":"244"},{"text":"2 cuill\u00e8res \u00e0 caf\u00e9de moutarde forte","isGood":false,"type":null,"points":"-4","id_qcm_response":"245"},{"text":"1 cuill\u00e8re \u00e0 soupe de moutarde forte","isGood":true,"type":null,"points":"10","id_qcm_response":"246"}]}}]}}

**Comment by maechler** [comment]
I have not used jstree in while now, but the code that caused this issue seems to be removed. You probably have an other issue than described here: https://groups.google.com/forum/#!topic/jstree/PsGONdCZml8

Is your jQuery version above 1.9.1? Do you see any output from the ajax request or the JavaScript console in the developer tools?

**Comment by clasie** [comment]
Hello,

   I use the jQuery JavaScript Library v1.10.2.  

   Thank you for the link but there is no 'window.ActiveXObject' in jstree.js

   I'll simplify the data structure and test again.

Kind regards.
Claude

**Comment by clasie** [comment]
Ok, I simplified drastically de json injected to the tree to:

'data': [
	{ "id": "170", "parent": "#", "text": "New questionnaire label", "**data**": { "TYPE": "QUESTIONNAIRE"} },
	{ "id": "171", "parent": "170", "text": "New question label", "**data**": { "TYPE": "QUESTION" } }
]

... and the strange thing is that if a leave the word "data",  the tree do not build,
if I change the word "data" to "datax" -> the tree loads with its child....

... but I need the "data" word to store information needed after when processing with the tree.
